### DISCUSSION TITLE
Dont get the point of permissions

### DISCUSSION THREAD

**Discussion by charlesooo** [tauri-apps/tauri#11476] [General]
permissions are pointless.
i tried for a day using API to write a file in c:/users/public, and could not do that.
end up using tauri::command and std::fs.
so, what is the point of these complex permission settings?

**Comment by maxbodin** [comment]
You're absolutely right! I’ve also been struggling with a similar issue in Tauri v2, trying to save files in specific directories.

Could you share more details about how you used `tauri::command` and `std::fs` to bypass these restrictions? I’d love to understand your approach and see if it can work for my use case as well. Thanks in advance!

**Reply by charlesooo** [reply]
```rust
use std::fs::write;
use tauri::ipc::{InvokeBody, Request};

#[tauri::command]
fn win_save_reg(request: Request) -> String {
    if let InvokeBody::Raw(data) = request.body() {
        if let Some(info) = APP_INFO.get() {
            let file = get_license_path(&info.name);
            if let Ok(()) = write(file, data) {
                return "OK".to_string();
            };
        }
    }
    "".to_string()
}

 tauri::Builder::default()
        .plugin(tauri_plugin_fs::init())
        .plugin(tauri_plugin_shell::init())
        .plugin(tauri_plugin_window_state::Builder::new().build())
        .invoke_handler(tauri::generate_handler![
            win_save_reg
        ])
```

for tauri2.0. 
command returns string as result to frontend.
i am learning rust, dont know if this is best practice.

**Comment by lojdt** [comment]
skill issue

**Reply by maxbodin** [reply]
That's not helpful....................

**Reply by charlesooo** [reply]
dev in rust is hard. this is making it harder

**Reply by FabianLars** [reply]
The permissions are not affecting the rust side at all though 😕
